flutter/lib/src/sentry_flutter.dart

```diff
@@ -35,27 +35,39 @@ mixin SentryFlutter {
       throw ArgumentError('OptionsConfiguration is required.');
     }
 
-    // It is necessary to initialize Flutter method channels so that
-    // our plugin can call into the native code.
-    WidgetsFlutterBinding.ensureInitialized();
-
     final flutterOptions = SentryFlutterOptions();
+
     // first step is to install the native integration and set default values,
     // so we are able to capture future errors.
-    _addDefaultIntegrations(
-      flutterOptions,
+    final defaultIntegrations = _createDefaultIntegrations(
       isIOSChecker,
       isAndroidChecker,
       packageLoader,
     );
 
+    if (appRunner != null) {
+      final runBefore = () => WidgetsFlutterBinding.ensureInitialized();
+      final runAfter = () => appRunner();
+      final rzgi = RunZoneGuardedIntegration(runBefore, runAfter);
+      flutterOptions.addIntegration(rzgi);
+
+      for (final defaultIntegration in defaultIntegrations) {
+        flutterOptions.addIntegration(defaultIntegration);
+      }
+    } else {
+      WidgetsFlutterBinding.ensureInitialized();
+      for (final defaultIntegration in defaultIntegrations) {
+        flutterOptions.addIntegration(defaultIntegration);
+      }
```

Comment:
the `for` loop could be just outside of the `if/else`, it'd give the same result